### PR TITLE
Add support for percentage step size to fans

### PR DESCRIPTION
**Pull request by bdraco**
## Proposed change

Add support for percentage step size to fans

yOrRMOWYmq

The pin value is now rounded to prevent floating points from showing up in the UI.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion: https://github.com/home-assistant/core/pull/46512
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by spacegaier** [comment]
Not having the pin feels a bit inconsistent since I think we have it in most (?) other sliders. Couldn't we just round the pin value?

**Comment by bdraco** [comment]
> Not having the pin feels a bit inconsistent since I think we have it in most (?) other sliders. Couldn't we just round the pin value?

Any idea how to do that?  I couldn't find a way that worked with the underlying api

**Comment by Misiu** [comment]
@spacegaier @bdraco maybe do it the other way:
Leave the step size=1, but change the slider max value. So if you have 7 speeds then you will move the value from 1 to 7.
This way the pin will show the speed as a number, but not as a percentage value (no need for rounding).

**Comment by bdraco** [comment]
While I think that will solve the pin issue, we would still need to translate the speed number back into a percentage so we still get a float, but at least we could hide it.

If the fan doesn't have a fixed step (100 speeds) aka variable rate fan they would get 1-100 still.

The downside is we loose a bit of the abstraction.

**Comment by bdraco** [comment]
I came up with a solution to avoid the need to hide the pin.

File: src/components/ha-slider.js [hunk 1]
```diff
@@ -79,6 +79,10 @@ class HaSlider extends PaperSliderClass {
     return subTemplate;
   }
 
+  _setImmediateValue(newImmediateValue) {
```
Comment (balloob) [inline src/components/ha-slider.js:82]: `<ha-slider>` is used in various contexts. Is this desired in all of them? 

CC @bramkragten

**Review by balloob — APPROVED** [review]

File: src/components/ha-slider.js (hunk 1 above)
Comment (bdraco) [reply src/components/ha-slider.js:82]: I looked through all the other places, and I think it should be ok. We would have had the same issue with unrounded floats being displayed in the UI if it was an issue elsewhere.

**Pull request merged**